src/main/java/org/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -1,26 +1,37 @@
 package org.bookstore.service.impl;
 
 import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.bookstore.dto.BookDto;
+import org.bookstore.dto.CreateBookRequestDto;
+import org.bookstore.mapper.BookMapper;
 import org.bookstore.model.Book;
 import org.bookstore.repository.BookRepository;
 import org.bookstore.service.BookService;
 import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto bookRequestDto) {
+        Book savedBook = bookRepository
+                .save(bookMapper.toModel(bookRequestDto));
+        return bookMapper.toDto(savedBook);
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto findById(Long id) {
+        return bookMapper.toDto(bookRepository.findById(id));
```

Comment:
when you adapt repository to return Optional, you can load book here via single chain:
```
return repo.findById(id)
              .map(mapper::map)
              .orElseThrow(...)
```